### PR TITLE
Banner targeting by page context

### PR DESCRIPTION
**Pull request by tomrf1**
SDC change: https://github.com/guardian/support-dotcom-components/pull/953

I've moved/renamed `epicTestTargetContentEditor.tsx`, and changed the props to take the new `PageContextTargeting` model

### CODE REVIEW

File: app/models/BannerTests.scala [hunk 1]
```diff
@@ -75,6 +75,7 @@ case class BannerTest(
     minArticlesBeforeShowingBanner: Int,
     userCohort: UserCohort,
     locations: List[Region] = Nil,
+    contextTargeting: Option[ContextTargeting] = Some(ContextTargeting(Nil,Nil,Nil,Nil)),
```
Comment (tomrf1) [inline app/models/BannerTests.scala:78]: existing banner tests will not have this field, so set a default with no targeting

File: app/models/PageContextTargeting.scala [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package models
+
+case class PageContextTargeting(
```
Comment (tomrf1) [inline app/models/PageContextTargeting.scala:3]: I have not updated the epic model to use this type as it already has the fields (`tagIds` etc) at the top level, so it'd require a data migration.
But I thought it was nice to put these fields together in a type that can be passed around in SAC + SDC

File: public/src/components/channelManagement/testEditorContextTargeting.tsx [hunk 3]
```diff
@@ -0,0 +1,76 @@
+import React from 'react';
```
Comment (tomrf1) [inline public/src/components/channelManagement/testEditorContextTargeting.tsx:1]: Most of this is copied from epicTestTargetContentEditor.tsx.
It now uses the `PageContextTargeting` model

File: app/models/BannerTests.scala [hunk 4]
```diff
@@ -75,6 +75,7 @@ case class BannerTest(
     minArticlesBeforeShowingBanner: Int,
     userCohort: UserCohort,
     locations: List[Region] = Nil,
+    contextTargeting: Option[PageContextTargeting] = Some(PageContextTargeting(Nil,Nil,Nil,Nil)),
```
Comment (tjmw) [inline app/models/BannerTests.scala:78]: Oh! I'm curious as to why the default here isn't `None` given that it's an `Option`?

**Review by tjmw — APPROVED** [review]

File: app/models/BannerTests.scala (hunk 4 above)
Comment (tomrf1) [reply app/models/BannerTests.scala]: Good question. I've just changed this to not be an Option.
We can use the default value to handle any records in dynamodb that do not have the field.
And the client can expect this field to always exist

**Pull request merged**